### PR TITLE
feat: 🎸 Chart series colors connected with legend

### PR DESCRIPTION
**Pull request by dariuszlacheta**

### CODE REVIEW

File: packages/charts/src/components/legend/legend.utils.ts [hunk 1]
```diff
@@ -88,13 +90,13 @@ export const createSliderTransition = (
     };
   return {
     initial: (direction: number) => ({
-      y: direction > 0 ? itemPositionNext : itemPositionPrevious,
+      y: direction > 0 ? itemPositionNext + 100 : itemPositionPrevious - 100,
```
Comment (apryka) [inline packages/charts/src/components/legend/legend.utils.ts:93]: need to discuss later on how to remove these magic numbers

File: packages/charts/src/charts/bar/components/bar/bar.component.tsx [hunk 2]
```diff
@@ -66,13 +68,17 @@ export const Bar: FC<Props> = ({
...
+  }, [dataSeriesOffset]);
+
   const commonProps = {
     height,
     width,
     fill: color,
     style: transitionStyle,
   };
-
+  console.log(colorOutOfRange);
```
[the remark this answers is not in the document]
Comment (dariuszlacheta) [reply packages/charts/src/charts/bar/components/bar/bar.component.tsx]: Removed

File: packages/charts/src/components/legend/legend.utils.ts (hunk 1 above)
Comment (dariuszlacheta) [reply packages/charts/src/components/legend/legend.utils.ts]: No magic there right now, clear math :)

File: packages/charts/src/charts/line/components/lines/lines.component.tsx [hunk 3]
```diff
@@ -141,6 +158,13 @@ const Lines = ({
     }
   }, [activeKey]);
 
+  useEffect(() => {
+    if (initialDrawFinished) {
+      lineControls.start(AnimationVariants.OffsetChange);
+      areaControls.start(AnimationVariants.OffsetChange);
```
Comment (maciejrybaniec) [inline packages/charts/src/charts/line/components/lines/lines.component.tsx:164]: IMO the `areaControls` animation should be used only for enabled `areaMode`. So we could enhance `useEffect` with additional dependency on `areaMode` and add proper condition.

**Review by maciejrybaniec — CHANGES_REQUESTED** [review]

File: packages/widgets/src/components/area/area-chart.widget.tsx [hunk 4]
```diff
@@ -30,6 +33,10 @@ export const AreaChartWidget: FC<Props> = ({
 }) => {
   const { disabledKeys, updateKeys } = useLegend();
   const [activeKey, setActiveKey] = useState(null);
+  const [dataSeriesOffset, setDataSeriesOffset] = useState<[number, number]>([
```
[the remark this answers is not in the document]
Comment (dariuszlacheta) [reply packages/widgets/src/components/area/area-chart.widget.tsx]: Changed

File: packages/charts/src/charts/line/components/lines/lines.component.tsx (hunk 3 above)
Comment (dariuszlacheta) [reply packages/charts/src/charts/line/components/lines/lines.component.tsx]: Changed

**Review by mczapkowicz — APPROVED** [review]

**Review by apryka — APPROVED** [review]

**Pull request merged**